### DISCUSSION TITLE
分布式的情况下 各个机器如何感知 是否有新向量插入呢

### DISCUSSION THREAD

**Discussion by bendanzzc** [milvus-io/milvus#19155] [Q&A and General discussion] [answered]
看了关于分布式的教程，说一台只写，其他只读，那比如我插入了新向量，其他只读机器如何感知到我插入了呢，如果按照文档的说明是1s一次flush，那只读的机器，岂不是要高频率的去访问磁盘查看是否有新的向量写入？

**Comment by yhmo** [comment, marked as answer]
似乎你说的是milvus 1.x版本，1.x版本有一个分布式方案叫Mishards，一写多读。执行flush时，写节点把数据写入存储(本地存储或是其他共享存储)，然后再MySQL中写入相应的元数据。proxy接到search请求时，会去元数据(也就是MySQL)中获取要访问哪些数据文件，然后告知读节点，读节点就知道要读取哪些数据了，如果数据已存在于读节点的内存中，那就不需要再次读取。
